nncf/common/utils/priority_queue.py

```diff
@@ -0,0 +1,67 @@
+"""
+ Copyright (c) 2022 Intel Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+from typing import List
+
+
+class PriorityQueue:
```

Comment:
For my understanding:

Why we can not use the priority queue from the [heapq](https://docs.python.org/3/library/heapq.html) module?

Reply:
`heqpq` module provides only min-heap, while I would like to use max-heap. The way how to adopt min-heap to max-heap is quite unfriendly.  I was surprised that standard python doesn't provide such an option.

Anyway, after some time I would agree that using the adopted `heapq` module is better. But it is a point for discussion

Reply:
I have one more question. Could you please explain why you need to use the priority queue? Why you can not use sorting by priority of algorithm inside the `apply()` method?

Reply:
Where does this code come from? Is it purely own implementation? Do we have tests for it?